ts/store/reducers/content.ts

```diff
@@ -52,8 +54,24 @@ const initialContentState: ContentState = {
 };
 
 // Selectors
-export const municipalitySelector = (state: GlobalState) =>
-  state.content.municipality;
+// return municipality metadata if codice catastale mathes with the one in profile fiscal code
+export const municipalitySelector = createSelector(
+  [(state: GlobalState) => state.content.municipality, profileSelector],
+  (municipality, profile): pot.Pot<MunicipalityMetadata, Error> => {
+    if (pot.isSome(municipality.codiceCatastale) && pot.isSome(profile)) {
+      const maybeCodiceCatastale = CodiceCatastale.decode(
+        profile.value.fiscal_code.substring(11, 15)
+      );
+      if (
+        maybeCodiceCatastale.isRight() &&
+        maybeCodiceCatastale.value === municipality.codiceCatastale.value
+      ) {
+        return municipality.data;
+      }
+    }
+    return pot.none;
+  }
+);
```

Comment:
Instead of manage the error in the selector, the error could be managed when the municipality request is sent, storing it in the app state itself.
 
A case such as the following could be added to the `content` state reducer: 
``````
case getType(contentMunicipalityLoad.failure):
      return {
        ...state,
        municipality: {
          codiceCatastale: pot.toError(
            state.municipality.codiceCatastale,
            action.payload
          ),
          data: pot.toError(state.municipality.data, action.payload)
        }
      };
``````
and `watchContentMunicipalityLoadSaga` could be changed such as:

````````
export function* watchContentMunicipalityLoadSaga(): Iterator<Effect> {
  yield takeEvery(getType(contentMunicipalityLoad.request), function*(
    action: ActionType<typeof contentMunicipalityLoad["request"]>
  ) {
    const codiceCatastale = action.payload;
    try {
      const response: SagaCallReturnType<
        typeof fetchMunicipalityMetadata
      > = yield call(
        fetchMunicipalityMetadata,
        contentClient.getMunicipality,
        codiceCatastale
      );

      if (response.isRight()) {
        yield put(
          contentMunicipalityLoad.success({
            codiceCatastale,
            data: response.value
          })
        );
      } else {
        yield put(
          contentMunicipalityLoad.failure(Error(response.value.message))
        );
      }
    } catch (e) {
      yield put(contentMunicipalityLoad.failure(e));
    }
  });
}

Reply:
@valentinacala For sure we are missing the failure handling. I will add it. 

Anyway the **municipalitySelector** doesn't handle the error loading, instead it checks if the store contains the required data, example: 

the store contains data about my CF **ABCDEF**. If I do logout without purging the store (first option on logout) this data is still stored into the store. If another account log-in into the app and asks for municipality the store will return the stored data (the data relative to the previous account **ABCDEF**). So the selector has to return the municipality data only if the requiring codiceCatastale is the same of the stored one.